### PR TITLE
Created a new crosslink file to store versioned links.

### PR DESCRIPTION
**Pull request by dcbenito**
I have added a new file to store links for cross-reference.

### CODE REVIEW

File: documentation/library_guide/common-cross-document-links.txt
Comment (akukanov) [inline documentation/library_guide/common-cross-document-links.txt:1]: Looks like this is a duplication of another file with a different name, and should be deleted?

File: documentation/library_guide/common_cross_document_links.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+CMake Support
```
Comment (akukanov) [inline documentation/library_guide/common_cross_document_links.txt:1]: "CMake Support" is rendered as a text on each HTML page, and so do other "headings" in this file. Either need to convert them to comments or simply remove, as these are not of much value, especially if the "macro" names are chosen to be self-descriptive.

File: documentation/library_guide/common_cross_document_links.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+DevGuide Introduction
+
+.. |dpcpp_gsg| replace:: Get Started with the Intel® oneAPI DPC++/C++ Compiler
+.. _dpcpp_gsg: https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/get-started-guide/2024-2/overview.html
+
+Get Started Introduction
+
+.. |dpcpp_gsg_2| replace:: Intel® oneAPI DPC++/C++ Compiler
+.. _dpcpp_gsg_2: https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/get-started-guide/2024-2/overview.html
```
Comment (akukanov) [inline documentation/library_guide/common_cross_document_links.txt:14]: I would rename as `|dpcpp_cpp_with_gsg_link|`

File: documentation/library_guide/common_cross_document_links.txt (hunk 1 above)
Comment (dcbenito) [reply documentation/library_guide/common_cross_document_links.txt:1]: Let's remove them. I had added them to keep track of where I was in the files, but I don't think that we need to do that moving forward.

File: documentation/library_guide/common_cross_document_links.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+.. |onedpl_main| replace:: Intel® oneAPI DPC++ Library Guide
+.. _onedpl_main: https://www.intel.com/content/www/us/en/docs/onedpl/developer-guide/2022-6/overview.html
```
Comment (akukanov) [inline documentation/library_guide/common_cross_document_links.txt:17]: `|onedpl_library_guide|`

File: documentation/library_guide/common_cross_document_links.txt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+Execution Policies
+
+.. |unroll_pragma| replace:: unroll Pragma
+.. _unroll_pragma: https://www.intel.com/content/www/us/en/docs/oneapi-fpga-add-on/developer-guide/2024-2/unroll-pragma.html
+
+.. |loop_analysis| replace:: Loop Analysis
+.. _loop_analysis: https://www.intel.com/content/www/us/en/docs/oneapi-fpga-add-on/developer-guide/2024-2/loop-analysis.html
+
+.. |fpga_hdbk| replace:: Intel® oneAPI FPGA Handbook
+.. _fpga_hdbk: https://www.intel.com/content/www/us/en/docs/oneapi-fpga-add-on/developer-guide/2024-2/intel-oneapi-fpga-handbook.html
```
Comment (akukanov) [inline documentation/library_guide/common_cross_document_links.txt:36]: `|fpga_handbook|`

File: documentation/library_guide/common_cross_document_links.txt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+.. |onedpl_main| replace:: Intel® oneAPI DPC++ Library Guide
+.. _onedpl_main: https://www.intel.com/content/www/us/en/docs/onedpl/developer-guide/2022-6/overview.html
+
+.. |yocto_layers| replace:: Layers for Yocto* Project
+.. _yocto_layers: https://www.intel.com/content/www/us/en/docs/oneapi-iot-toolkit/get-started-guide-linux/2024-0/adding-oneapi-components-to-yocto-project-builds.html
+
+Macros
+
+.. |vector| replace:: vector page in the Intel® oneAPI DPC++/C++ Compiler Developer Guide and Reference
+.. _vector: https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-2/vector.html
```
Comment (akukanov) [inline documentation/library_guide/common_cross_document_links.txt:25]: `|vector_pragma|`

File: documentation/library_guide/common_cross_document_links.txt (hunk 2 above)
Comment (dcbenito) [reply documentation/library_guide/common_cross_document_links.txt:14]: Approved.

File: documentation/library_guide/common_cross_document_links.txt (hunk 3 above)
Comment (dcbenito) [reply documentation/library_guide/common_cross_document_links.txt:17]: Approved.

File: documentation/library_guide/common_cross_document_links.txt (hunk 4 above)
Comment (dcbenito) [reply documentation/library_guide/common_cross_document_links.txt:36]: Approved.

File: documentation/library_guide/common_cross_document_links.txt (hunk 5 above)
Comment (dcbenito) [reply documentation/library_guide/common_cross_document_links.txt:25]: Approved.

File: documentation/library_guide/cmake_support.rst [hunk 6]
```diff
@@ -17,7 +17,7 @@ Requirements
 ============
 The minimal supported CMake version for |onedpl_short| is 3.11 on Linux and 3.20 on Windows.
 
-The supported `CMake Generator <https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#ninja-generators>`_ for Linux is `Unix Makefiles <https://cmake.org/cmake/help/latest/generator/Unix%20Makefiles.html>`_ (default). In the Windows environment, the supported generator is `Ninja <https://cmake.org/cmake/help/latest/generator/Ninja.html>`_ as described in the `Intel® oneAPI DPC++/C++ Compiler Developer Guide and Reference <https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/current/use-cmake-with-the-compiler.html>`_ which may be specified via ``-GNinja``.
+The supported `CMake Generator <https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#ninja-generators>`_ for Linux is `Unix Makefiles <https://cmake.org/cmake/help/latest/generator/Unix%20Makefiles.html>`_ (default). In the Windows environment, the supported generator is `Ninja <https://cmake.org/cmake/help/latest/generator/Ninja.html>`_ as described in the |cmake_support_page|_ which may be specified via ``-GNinja``.
```
Comment (akukanov) [inline documentation/library_guide/cmake_support.rst:20]: Split to several lines:
```suggestion
The supported `CMake Generator <https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#ninja-generators>`_
for Linux is `Unix Makefiles <https://cmake.org/cmake/help/latest/generator/Unix%20Makefiles.html>`_ (default). In the Windows
environment, the supported generator is `Ninja <https://cmake.org/cmake/help/latest/generator/Ninja.html>`_
as described in the |cmake_support_page|_ which may be specified via ``-GNinja``.
```

**Review by dcbenito — COMMENTED** [review]
Changes LGTM.

**Review by akukanov — APPROVED** [review]

**Pull request merged**